tools/metadata_viewer/kvstore.py

```diff
@@ -144,14 +144,17 @@ def __init__(self, snapshot_file_path):
         self.data = None
 
     def read(self):
-        with open(self.path, "rb") as f:
-            data = f.read(SNAP_HDR_SIZE)
-            if len(data) == SNAP_HDR_SIZE:
-                self.header = SnapshotHeader(
-                    *struct.unpack(SNAP_HDR_FMT, data))
-            if self.header:
-                self.meta = f.read(self.header.metadata_size)
-                self.data = f.read()
+        try:
```

Comment:
I think it's correct for a Snapshot pointing to a nonexistent path to raise an exception when trying to read it (we shouldn't suppress that).

The question I would ask is how are we constructing a Snapshot object that points to a nonexistent path?

Reply:
I've just formed a 3 nodes cluster locally and put some records. When I've used the metadata_viewer to poke around the contents themselves, I hit the issue since some of the directories didn't store any data yet. 

In a production deployment where enough number of records have been produced, I don't think this is the case. Do you think this is still worth to address?

Reply:
Got it. Thanks!